dso-l2/src/main/java/com/tc/services/TerracottaServiceProviderRegistryImpl.java

```diff
@@ -110,4 +112,21 @@ private void registerNewServiceProvider(ServiceProvider service) {
     logger.info("Initializing " + service);
     serviceProviders.add(service);
   }
+
+  @Override
+  public void dumpStateTo(StateDumper stateDumper) {
+    for (ServiceProvider serviceProvider : serviceProviders) {
+      if(serviceProvider instanceof StateDumpable) {
```

Comment:
Again, we probably want some comment here to explain that this is opt-in, for the moment.